src/mscorlib/src/System/Random.cs

```diff
@@ -114,11 +121,39 @@ public Random()
           return retVal;
       }
 
+
+      [ThreadStatic]
+      private static Random t_threadRandom;
+      private static Random s_globalRandom = new Random(GenerateGlobalSeed());
+
+      /*=====================================GenerateSeed=====================================
+      **Returns: An integer that can be used as seed values for consecutively
+                 creating lots of instances on the same thread within a short period of time.
+      ========================================================================================*/
+      private static int GenerateSeed() {
+          if (t_threadRandom == null) {
+              int seed;
+              lock (s_globalRandom) {
+                  seed = s_globalRandom.Next();
+              }
+              t_threadRandom = new Random(seed);
+          }
+          return t_threadRandom.Next();
+      }
+
+      /*==================================GenerateGlobalSeed====================================
+      **Action:  Creates a number to use as global seed.
+      **Returns: An integer that is safe to use as seed values for thread-local seed generators.
+      ==========================================================================================*/
+      private static int GenerateGlobalSeed() {
+          return Guid.NewGuid().GetHashCode();
```

Comment:
> additional computations

I have quantified those to 2.92 micro seconds. This is not "a bunch", it is a tiny amount. Are you concerned about 2.92 micro seconds?

Let's solve this problem for good and play it safe. Hopefully this is the last time this code needs to be touched.

> There is only one AppDomain in CoreCLR so that should not be an issue.

I understand CoreCLR code eventually makes its way into the Desktop product, does it not?